code/go/internal/validator/semantic/validate_input_policy_template_template_path.go

```diff
@@ -0,0 +1,100 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package semantic
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/elastic/package-spec/v3/code/go/internal/fspath"
+	"github.com/elastic/package-spec/v3/code/go/pkg/specerrors"
+)
+
+const (
+	inputPackageType packageType = "input"
+)
+
+var (
+	errRequiredTemplatePath  = errors.New("template_path is required for input type packages")
+	errFailedToReadManifest  = errors.New("failed to read manifest")
+	errFailedToParseManifest = errors.New("failed to parse manifest")
+	errTemplateNotFound      = errors.New("template file not found")
+	errInvalidPackageType    = errors.New("invalid package type")
+)
+
+type packageType string
+
+type inputPolicyTemplate struct {
+	Name         string `yaml:"name"`
+	TemplatePath string `yaml:"template_path"` // input type packages require this field
+}
+
+type inputPackageManifest struct { // package manifest
+	Type            packageType           `yaml:"type"`
```

Comment:
Could it be used `string` here instead of a custom type ?

```suggestion
	Type            string           `yaml:"type"`
```